src/Persistence/PersistentObjectFactory.php

```diff
@@ -364,26 +376,43 @@ protected function normalizeCollection(string $field, FactoryCollection $collect
      *
      * @internal
      */
-    protected function normalizeObject(object $object): object
+    protected function normalizeObject(string $field, object $object): object
     {
         $configuration = Configuration::instance();
 
-        if (
-            !$this->isPersisting()
-            || !$configuration->isPersistenceAvailable()
-        ) {
+        $object = unproxy($object, withAutoRefresh: false);
+
+        if (!$configuration->isPersistenceAvailable()) {
             return $object;
         }
 
-        $object = unproxy($object, withAutoRefresh: false);
-
         $persistenceManager = $configuration->persistence();
+
         if (!$persistenceManager->hasPersistenceFor($object)) {
             return $object;
         }
 
+        $inverseRelationship = $persistenceManager->inverseRelationshipMetadata(static::class(), $object::class, $field);
+
+        if ($inverseRelationship instanceof OneToOneRelationship) {
+            $this->tempAfterInstantiate[] = static function(object $newObject) use ($object, $inverseRelationship) {
+                try {
+                    set($object, $inverseRelationship->inverseField(), $newObject);
+                } catch (\Throwable) {
+                }
+            };
+        }
```

Comment:
```php
$address = static::addressFactory()
            ->afterInstantiate(
                static function(Address $a): void {
                    static::contactFactory()->create(['address' => $a]);
                }
            )->create();

self::assertNotNull($address->getContact());
```

it was a fluke that this kind of code did actually work. Not really sure why, maybe thanks to some doctrine lazy loading behavior. Now we're only flushing once here, we need to hydrate manually the inverse field